tests/Normalization/NormalizationGroupsProviderByStringTest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace DetailTest\Apigility\Normalization;
+
+use Detail\Apigility\Normalization\NormalizationGroupsProviderByString;
+use PHPUnit\Framework\TestCase;
+
+class NormalizationGroupsProviderByStringTest extends TestCase
+{
+    public function provideGroupsAsString(): array
+    {
+        return [
+            [
+                'foo',
+                ['Default', 'foo'],
+            ],
+            [
+                'foo,bar',
+                ['Default', 'foo', 'bar'],
+            ],
+            [ // Sanitation test
+                'foo , bar, foo-bar ,foo.bar,foo_bar,4foo.2bar',
+                ['Default', 'foo', 'bar', 'foobar', 'foo.bar', 'foo_bar', '4foo.2bar'],
```

Comment:
Dashes should be allowed in my opinion.

And the examples are not very good to actually test the sanitation.